BrainPortal/app/controllers/groups_controller.rb

```diff
@@ -268,6 +271,116 @@ def license_redirect #:nodoc:
+
+    license_text = params[:license_text]
+    cb_error 'Empty licenses are presently not allowed' if license_text.blank?
+
+    timestamp  = Time.zone.now.strftime("%Y-%m-%dT%H:%M:%S")
+    group_name = @group.name.gsub(/[^\w]+/, "")
+    file_name  = "license_#{group_name}_#{timestamp}.txt"
+    license    = @group.register_custom_license(license_text, current_user, file_name)
+    current_user.signs_license_for_project(license, @group, model='group')
+
+    flash[:notice] = 'A license is added. You can force users to sign multiple license agreements if needed.'
+    redirect_to :action => :show
+  end
+
+  def show_license #:nodoc:
+    # @group            =  @current_user.visible_groups.where(id: params[id]).first
+
+    unsigned_licenses = current_user.unsigned_custom_licenses(@group)
+
+    if unsigned_licenses.empty?
+      if @current_licenses.present?
+        flash[:notice] = 'You already signed all licenses'
+      else
+        flash[:notice] = 'No licenses are defined for this project'
+        redirect_to :action => :show
+        return
+      end
+    end
+
+    # What to show. If a license is given in params,
+    # we make sure it's a registered one and we pick that.
+    @license_id = false unless @current_licenses.include? @license_id
+    # If no valid license was given and there are unsigned licenses, pick the first
+    @license_id ||= unsigned_licenses.first.try(:to_i)
+    # Otherwise, show the first license.
+    @license_id ||= @current_licenses.first
+
+    # Load the text of the license
+    userfile = Userfile.find(@license_id)
+    userfile.sync_to_cache
+    @license_text = userfile.cache_readhandle { |fh| fh.read }
+
+    # Identifies if the current user has already signed it,
+    # and whether they are the author
+  end
+
+  def sign_license #:nodoc:
+
+    unless @group.custom_license_agreements.include?(@license_id)
+      flash[:error] = 'You are trying to access unrelated license. Try again or report the issue to the support.'
+      redirect_to :action => :show
+      return
+    end
+
+    if @is_signed
+      flash[:error] = 'You have already signed this license.'
+      redirect_to :action => :show
+      return
+    end
+
+    unless params.has_key?(:agree)
+      flash[:error] = "You cannot access that project without signing the License Agreement first."
+      redirect_to :action => :index
+      return
+    end
+
+    if params[:license_check].blank? || params[:license_check].to_i == 0
+      flash[:error] = "There was a problem with your submission. Please read the agreement and check the checkbox."
+      redirect_to show_license_group_path(@group)
+      return
+    end
+
+    license = Userfile.find(@license_id)
+    current_user.signs_license_for_project(license, @group, model='group')
+
+    if current_user.unsigned_custom_licenses(@group).empty?
+      flash[:notice] = 'You signed all the project licenses'
+      redirect_to :action => :show, :id => @group.id
```

Comment:
You have the same `redirect_to` in the `if` and in the `else`. Is it normal ? 

If it is ok should the `redirect_to` be outside of the `if ... else ... ` ?

Reply:
done